docs/readmes/proposals/p006_subscriber_state_view.md

```diff
@@ -0,0 +1,166 @@
+---
+id: p006_subscriber_state_view
+title: Displaying Run-time Subscriber State in NMS 
+hide_title: true
+---
+*Feature owner: @themarwhal*
+
+# Subscriber State Reporting From GW → Orc8r
+
+## Summary of currently available run-time subscriber state
+
+Currently, our view of subscribers is fairly fragmented over states reported by various services and event logs.
+
+### DirectoryD (IMSI → SessionID/MSISDN)
+
+* On session creation, IMSI to SessionID/MSISDN is added as an entry by SessionD. However, this is not ideal as SessionID is only unique per session not IMSI. This means that in a multi APN scenario, the first entry gets overwritten when the second session is initialized.
+* For CWF only, we also have an additional entry in DirectoryD that maps from IMSI→UE IPv4 added by PipelineD.
+
+### MobilityD (IMSI+APN → IPv4)
+
+* MobilityD also reports internal state to the orc8r from which we parse out IMSI+APN ↔ IPv4 mapping
+
+
+
+## Summary of currently available subscriber/policy config
+
+For non-Federated deployments, all subscriber and subscriber->policy configurations are made on the Orc8r. So the Orc8r at least has a sense of the desired state.
+For federated deployments where the subscriber configurations are federated to HSS or the policy configurations are federated to PCRF, the Orc8r does not have any visibility into the configurations received.
+
+
+## General Goal
+
+The goal here is to export consolidated session state owned by SessionD to the Orc8r so that we can 
+
+1. View all existing session states, uniquely identified by IMSI+APN 
+2. View runtime state of active policies + QoS
+
+
+
+## Proposed State from SessionD → GW State service
+
+Here is the state Protobuf type:
+
+```
+message State {
+    // Type determines how the value is deserialized and validated on the cloud service side
+    string type = 1; // 'subscriber_state'
+    string deviceID = 2; // IMSI'
+    // Value contains the operational state json-serialized.
+    bytes value = 3; // JSON serialized Subscriber State type below
+    uint64 version = 4; // ?
+}
+```
+
+Here is the blob value:
+
+```
+  subscriber_state:
+    type: object
+    description: An object that maps IMSI->APN->SubscriberState
+    properties:
+      imsi:
```

Comment:
Is `subscriber_state` and `session_state` mirroring any existing definitions? Because e.g. the `imsi` property is superfluous within the state service since it's the state's key

But if the `imsi` property is included because it's part of an existing definition, then that's fine